### PR TITLE
DPE-4613 Test for multi-relation scale in/out

### PR DESCRIPTION
**Pull request by paulomach**
Issue #432 is not easily reproducible.

This PR tries to reproduce the use case, using multiple applications connected to a single cluster.

### CODE REVIEW

**Comment by taurus-forever** [comment]
@paulomach The test passed, and looks solid.
Proposal: deploy mysql db 3 units and random(1-3) units of the router.

Mo IPs + more events inside.

File: tests/integration/test_multi_relations.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Build the charm and deploy 1 units to ensure a cluster is formed."""
+    # Build and deploy charm from local source folder
+    db_charm = await ops_test.build_charm(".")
+
+    config = {"profile": "testing"}
```
Comment (paulomach) [inline tests/integration/test_multi_relations.py:22]: @taurus-forever

**Comment by paulomach** [comment]
> @paulomach The test passed, and looks solid. Proposal: deploy mysql db 3 units and random(1-3) units of the router.
> 
> More IPs + more events inside.
> 
> Also, is profile=production used on purpose here?

It's on `testing`. 

I was out to replicate sunbeam deployment size + simultaneous scale in/out to catch their issue, but it may be helpful to exaggerate it here.

File: tests/integration/test_multi_relations.py (hunk 1 above)
Comment (taurus-forever) [reply tests/integration/test_multi_relations.py:23]: Right... I expected to see it as line 34. Missed. Tnx!

File: tests/integration/test_multi_relations.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        series="jammy",
+        trust=True,
+    )
+
+    for i in range(1, SCALE_OUT + 1):
+        await ops_test.model.deploy(
+            "mysql-test-app",
+            application_name=f"app{i}",
+            num_units=1,
+            channel="latest/edge",
```
Comment (taurus-forever) [inline tests/integration/test_multi_relations.py:41]: auto_start_writes=false ? re: performance of GH Runners.... 
P.S. consider to add rate-limit for mysql-test-app to do NOT kill DB in such cases: https://github.com/canonical/postgresql-test-app/blob/main/config.yaml#L11-L15

So you will be able to use here `sleep_interval=1000(ms)` to rate-limit 1 insert per second from each test-app.

**Review by taurus-forever — APPROVED** [review]
LGTM. I would also use 3 units for router (+ some test app also for 3 units.)

One performance recomentation here due to Inserts auto-start on test-app.

**Review by shayancanonical — APPROVED** [review]

**Comment by shayancanonical** [comment]
May be we can add a couple more `mysql-test-app` applications and `router` applications to see if we run into https://github.com/canonical/mysql-k8s-operator/issues/476. Thoughts?

**Pull request merged**